### DISCUSSION TITLE
Poco::Thread setName

### DISCUSSION THREAD

**Discussion by micheleselea** [pocoproject/poco#4471] [Q&A]
Hi all, is there a good reason why now the Thread::setName call a setNameImpl that has a POCO_MAX_THREAD_NAME_LEN of 15 and truncate thread names? I think it's the same in POSIX and WIN32

**Comment by aleks-f** [comment]
> The name can be up to 16 bytes long, including the terminating null byte.

https://man7.org/linux/man-pages/man2/prctl.2.html

**Comment by micheleselea** [comment]
Yes correct, but I think that the OS ThreadName and the member ThreadName can be different. I mean, I think there's no reason to limit the member class name because the os name is limited, or perhaps we can keep both name and osName
